### ISSUE TITLE
Expose charmc's moduleinit functionality

### ISSUE THREAD

**Issue by nilsleiffischer** [UIUC-PPL/charm#3210]
tldr: Please expose the last(?) remaining piece of private functionality in `charmc`, i.e. the code related to the "module initialization functions".

**Details:**

In our code that depends on Charm++ we are currently moving away from wrapping our compiler with `charmc` because it doesn't play well with a CMake build system. In our CMake build system we define Charm++ as an "imported target" and associate include directories, libraries and compiler flags with that target so they propagate automatically to all of our libraries and executables that link with it [1]. All of this information we can retrieve from a call to `charmc -print-building-blocks`, so it's great to have that functionality in `charmc`! However, `charmc` still does one thing (or perhaps more?) that isn't exposed and that we have to work around in our build system: It generates, compiles and links a file with "module initialization functions" (see https://github.com/UIUC-PPL/charm/blob/e71239838c360dfebe7c31849651d45d1c02dd36/src/scripts/charmc#L2246). AFAICT these functions are usually empty, but are needed to link a charm program. It would be great if this could be made a public and documented interface so we can reliably work with it. This could be the last remaining piece to make the `charmc` compiler wrapper optional.

[1] I can recommend this post for an (opinionated) view on modern CMake: https://pabloariasal.github.io/2018/02/19/its-time-to-do-cmake-right/
